### DISCUSSION TITLE
Docker improvements

### DISCUSSION THREAD

**Discussion by APraxx** [SamTV12345/PodFetch#150] [General]
hi, 

i think the dockerfile can be enhanced a bit.
- drop root
     It seems the current docker image runs the service as root. this should be dropped down to a regular user if not needed.

- maybe give the option to specify a user pid/gid

i could make a pr if you like

**Comment by SamTV12345** [comment]
Hi,

sure please do so. I don't know if the standard docker image with distroless needs that but the ubuntu image could definitely benefit from that.

**Comment by APraxx** [comment]
So i did some research because i wanted this to be the same across all the images.
But after digging into it I'm back at the start.

is there a reason to not just forcing the user via docker at runtime?

basically just  `user: ${UID:-1000}:${GID:-1000}` in compose

reason: everything else ranges from a simple USER in the dockerfile to scripting for runtime changes

**Comment by SamTV12345** [comment]
I guess not. That would be the easiest solution. Do you want to create a pr?

**Reply by APraxx** [reply]
done

**Comment by APraxx** [comment]
@SamTV12345 other thing i noticed and is indirect a docker issue.
your dev? branch has still podgrabv2 as it's name.

this kinda [leaks](https://github.com/SamTV12345/PodFetch/blob/e0d3fbbb5627682243d40643389a04386b750b64/README.md?plain=1#L70) into the main branch

**Comment by APraxx** [comment]
would you be open for a [ghcr.io](https://ghcr.io) images? this could be done via github actions and could for a start track master

**Comment by SamTV12345** [comment]
Are you good at github actions? I must admit I have never been that comfortable with that syntax etc. I would like to have an action that runs on master and on the pull requests.The pull requests should check if the build even works.

**Reply by APraxx** [reply]
ill look into it

**Reply by SamTV12345** [reply]
Thanks for the help.

**Reply by APraxx** [reply]
build on every commit to main?

**Reply by SamTV12345** [reply]
On release of a tag v** would be enough. Normally versions are v3.5.1 etc.
